src/ir_Technibel.cpp

```diff
@@ -0,0 +1,416 @@
+// Copyright 2020 Quentin Briollant
+
+/// @file
+/// @brief Support for Technibel protocol.
+
+#include "ir_Technibel.h"
+#include "IRrecv.h"
+#include "IRsend.h"
+#include "IRtext.h"
+#include "IRutils.h"
+#include <algorithm>
+
+using irutils::addBoolToString;
+using irutils::addModeToString;
+using irutils::addFanToString;
+using irutils::addLabeledString;
+using irutils::addTempToString;
+using irutils::minsToString;
+using irutils::setBit;
+using irutils::setBits;
+
+const uint16_t kTechnibelAcHdrMark = 8836;;
+const uint16_t kTechnibelAcHdrSpace = 4380;;
+const uint16_t kTechnibelAcBitMark = 523;;
+const uint16_t kTechnibelAcOneSpace = 1696;;
+const uint16_t kTechnibelAcZeroSpace = 564;;
+const uint32_t kTechnibelAcGap = kDefaultMessageGap;
+const uint16_t kTechnibelAcFreq = 38000;
+const uint16_t kTechnibelAcOverhead = 3;
+
+
+#if SEND_TECHNIBEL_AC
+// Send an Technibel AC formatted message.
+//
+// Args:
+//   data:   The message to be sent.
+//   nbits:  The number of bits of the message to be sent.
+//           Typically kTechnibelAcBits.
+//   repeat: The number of times the command is to be repeated.
+//
+// Status: STABLE / Reported as working on a real device.
+void IRsend::sendTechnibelAc(const uint64_t data, const uint16_t nbits,
+                            const uint16_t repeat) {
+  sendGeneric(kTechnibelAcHdrMark, kTechnibelAcHdrSpace,
+              kTechnibelAcBitMark, kTechnibelAcOneSpace,
+              kTechnibelAcBitMark, kTechnibelAcZeroSpace,
+              kTechnibelAcBitMark, kTechnibelAcGap,
+              data, nbits, kTechnibelAcFreq, false,  // LSB First.
+              repeat, kDutyDefault);
+}
+#endif  // SEND_TECHNIBEL_AC
+
+#if DECODE_TECHNIBEL_AC
+// Decode the supplied TECHNIBEL_AC message.
+//
+// Args:
+//   results: Ptr to the data to decode and where to store the decode result.
+//   offset:  The starting index to use when attempting to decode the raw data.
+//            Typically/Defaults to kStartOffset.
+//   nbits:   The number of data bits to expect. Typically kTechnibelAcBits.
+//   strict:  Flag indicating if we should perform strict matching.
+// Returns:
+//   boolean: True if it can decode it, false if it can't.
+//
+// Status: STABLE / Expected to be working.
```

Comment:
Ditto earlier doxygen-style comments comment.
```
/// Decode the supplied Technibel A/C message.
/// Status: STABLE / Expected to be working
/// @param[in,out] results Ptr to the data to decode & where to store the decode
/// @param[in] offset The starting index to use when attempting to decode the
///   raw data. Typically/Defaults to kStartOffset.
/// @param[in] nbits The number of data bits to expect. Typically kTechnibelAcBits.
/// @param[in] strict Flag indicating if we should perform strict matching.
/// @return A boolean. True if it can decode it, false if it can't.
```